### PR TITLE
Add commentator related resources to Avo

### LINKED ISSUE #7259: Feature Add Commentator related resources to Avo
### Is your feature request related to a problem? Please describe

This feature request follows #6733 as the base and is for adding the commentator related resources to Avo dashboard. This includes comments, commentator_thread, commentator_comment, subscription resources.

### Describe the solution you'd like

Using the RailsAdmin dashboard as reference the following mentioned resources are added to Avo dashboard.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### Are you working on this?

Yes

### PR DESCRIPTION
**Pull request by ShinichiShi**
Fixes #7259

 

#### Describe the changes you have made in this PR -
This PR adds the resources responsible for adding/editing comments in the admin dashboard. Currently these were present in the RailsAdmin panel and was required to be moved to Avo
 
### Screenshots of the UI changes (If any) -

image

---

## Code Understanding and AI Usage

**Did you use AI assistance (ChatGPT, Claude, Copilot, etc.) to write any part of this code?**
- [x] No, I wrote all the code myself
- [ ] Yes, I used AI assistance (continue below)

**If you used AI assistance:**
- [ ] I have reviewed every single line of the AI-generated code
- [ ] I can explain the purpose and logic of each function/component I added
- [ ] I have tested edge cases and understand how the code handles them
- [ ] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**

 These resources were already present in the railsAdmin console, and were added into Avo with the same fields. 

---

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [x] I have performed a self-review of my code
- [x] **I can explain the purpose of every function, class, and logic block I added**
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [ ] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

---

Note: Please check **Allow edits from maintainers** if you would like us to assist in the PR.

## Release Notes

* **New Features**
  * Added admin management interface for comments with searchable creator/editor associations, content moderation, and vote metrics.
  * Added admin management interface for discussion threads with closure tracking and comment/subscription counting capabilities.
  * Added admin management interface for discussion subscriptions with subscriber and thread associations.

### CODE REVIEW

File: app/avo/resources/comment.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    field :body, as: :textarea, required: true
+    field :deleted_at, as: :date_time
+    field :cached_votes_up, as: :number
+    field :cached_votes_down, as: :number
+
+    field :created_at, as: :date_time, readonly: true, sortable: true, hide_on: %i[edit new]
+    field :updated_at, as: :date_time, readonly: true, hide_on: %i[edit new]
+
+    field :children, as: :has_many
```
[the remark this answers is not in the document]
Comment (ShinichiShi) [reply app/avo/resources/comment.rb]: fixed
